lib/publish.js

```diff
@@ -30,53 +30,78 @@ module.exports = async (pluginConfig, context) => {
   debug('milestones: %o', milestones);
 
   if (assets && assets.length > 0) {
-    const globbedAssets = await getAssets(context, assets);
+    // Skip glob if url is provided
+    const urlAssets = assets.filter(asset => asset.url);
+    debug('url assets: %o', urlAssets);
+    const globbedAssets = await getAssets(
+      context,
+      assets.filter(asset => !asset.url)
+    );
     debug('globbed assets: %o', globbedAssets);
+    const allAssets = [...urlAssets, ...globbedAssets];
+    debug('all assets: %o', allAssets);
 
     await Promise.all(
-      globbedAssets.map(async asset => {
+      allAssets.map(async asset => {
         const {path, label, type, filepath} = isPlainObject(asset) ? asset : {path: asset};
-        const file = resolve(cwd, path);
-        let fileStat;
-
-        try {
-          fileStat = await stat(file);
-        } catch (_) {
-          logger.error('The asset %s cannot be read, and will be ignored.', path);
-          return;
-        }
-
-        if (!fileStat || !fileStat.isFile()) {
-          logger.error('The asset %s is not a file, and will be ignored.', path);
-          return;
-        }
-
-        debug('file path: %o', path);
-        debug('file label: %o', label);
-        debug('file type: %o', type);
-        debug('file filepath: %o', filepath);
-
-        // Uploaded assets to the project
-        const form = new FormData();
-        form.append('file', createReadStream(file));
-
-        const uploadEndpoint = urlJoin(gitlabApiUrl, `/projects/${encodedRepoId}/uploads`);
-
-        debug('POST-ing the file %s to %s', file, uploadEndpoint);
-
-        let response;
-        try {
-          response = await got.post(uploadEndpoint, {...apiOptions, body: form}).json();
-        } catch (error) {
-          logger.error('An error occurred while uploading %s to the GitLab project uploads API:\n%O', file, error);
-          throw error;
+        let _url = asset.url;
+        if (_url) {
+          // Allow loading variable from process.env load CI_JOB_ID for attaching CI artifact
+          const matchedVariables = _url.match(/\${[a-zA-Z].*}/);
+          if (matchedVariables) {
+            for (const matchedVariable of matchedVariables) {
+              const variableName = matchedVariable.slice(2, -1);
+              if (process.env[variableName]) {
+                _url = _url.replace(matchedVariable, process.env[variableName]);
```

Comment:
This will only find the first usage.
I do not know if somebody might do this, but if one variable might be used multiple times, it will not be replaced.

Maybe:
```suggestion
                _url = _url.replace(new RegExp('\\' + matchedVariable, 'g), process.env[variableName]);
```
The `\\` are needed to escape `$`.

@awcjack What do you think?

Reply:
Fixed the global flag first

Reply:
Using escapeRegExp function now for escaping all special symbol in regexp

Reply:
```
function escapeRegExp(string) {
  return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'); // $& means the whole matched string
}
```